lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -83,6 +84,11 @@ def render(engine, data, *)
 
             tracer = Datadog.configuration[:sinatra][:tracer]
 
+            if Datadog.configuration[:sinatra][:distributed_tracing]
+              context = HTTPPropagator.extract(request.env)
+              tracer.provider.context = context if context.trace_id
```

Comment:
Not sure about that. Sinatra can play with the Rack integration, and if `c.use :rack` is activated, our Context should have already a Rack span. If we do this change without checking, we may lose data here. I don't know why we can't automatically activate the Rack patch like we did for Rails: https://github.com/DataDog/dd-trace-rb/blob/f6e83cac35cd67b9042fe3cbca69dd85be933732/lib/ddtrace/contrib/rails/framework.rb#L28-L33

Anyway, since we didn't use the Rack integration under the hood, the approach that doesn't break anything is checking if we have spans before replacing the context. In that case, I would say to not do anything.

Side-effects:
* if you're using Rack, Rack should activate `distributed_tracing`
* if you're using something else on top of Sinatra, well it should be a supported framework to enable distributed tracing

What do you think?

Reply:
After our discussion, it's a matter of tradeoffs. Since Sinatra doesn't use Rack under the hood, we should keep this logic. The only problem with this approach, is that you may detach Sinatra and Rack traces, if you enable both and activate Sinatra distributed tracing.

The behavior is supported if you activate `distributed_tracing` only on Rack, that makes sense and is the lesser error prone approach. Let's keep it that way.